### PR TITLE
fix(runtime): respect 'rtp' order for all runtime files

### PR DESCRIPTION
**Pull request by zeertzjq**

### CODE REVIEW

File: src/nvim/path.c [hunk 1]
```diff
@@ -563,11 +562,9 @@ bool path_has_exp_wildcard(const char *p)
       p++;
       continue;
     }
+#endif
 
     const char *wildcards = "*?[{";
-#else
-    const char *wildcards = "*?[";  // Windows.
-#endif
```
Comment (zeertzjq) [inline src/nvim/path.c:570]: This change makes behavior more consistent, considering the fact that `{` does have a special meaning on Windows when there are with wildcards.

File: runtime/ftplugin.vim [hunk 2]
```diff
@@ -30,10 +30,7 @@ augroup filetypeplugin
       for name in split(s, '\.')
         " Load Lua ftplugins after Vim ftplugins _per directory_
         " TODO(clason): use nvim__get_runtime when supports globs and modeline
-        exe printf('runtime! ftplugin/%s.vim ftplugin/%s.lua
-                           \ ftplugin/%s_*.vim ftplugin/%s_*.lua
-                           \ ftplugin/%s/*.vim ftplugin/%s/*.lua',
-                           \ name, name, name, name, name, name)
+        exe $'runtime! ftplugin/{name}.{{vim,lua}} ftplugin/{name}_*.{{vim,lua}} ftplugin/{name}/*.{{vim,lua}}'
```
[the remark this answers is not in the document]
Comment (zeertzjq) [reply runtime/ftplugin.vim:33]: `:h $'`

[the remark this answers is not in the document]
Comment (clason) [reply runtime/ftplugin.vim:33]: Ah, only got ported this year (by you, who else ;)). Good to know!

File: test/functional/legacy/097_glob_path_spec.lua [hunk 3]
```diff
@@ -28,47 +28,25 @@ describe('glob() and globpath()', function()
     -- Consistent sorting of file names
     command('set nofileignorecase')
 
-    if helpers.is_os('win') then
```
Comment (zeertzjq) [inline test/functional/legacy/097_glob_path_spec.lua:31]: This test is mostly the same as `Test_glob()` in `test_escaped_glob.vim`, which is skipped on non-Unix.

**Comment by zeertzjq** [comment]
Hmm, "{" is included in 'isfname' by default on Windows. Need to rethink about this then.

**Pull request merged**